### PR TITLE
Fix absolute path

### PR DESCRIPTION
**Pull request by sarangan12**
@vishrutshah @veronicagg @salameer @devigned @NiklasGustafsson Please review and approve

Tested locally against the SDK.

Related to: #730. Please refer the issue for more context.

Comparison of Ruby SDK befor & after changes are available at: \\saran-dev1\shared\PR2059

To see how node sdk handles the task, refer: https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/cdnManagement/lib/cdnManagementClient.js#L160

### CODE REVIEW

File: src/generator/AutoRest.Ruby/Templates/MethodTemplate.cshtml [hunk 1]
```diff
@@ -120,7 +120,7 @@ def @(Model.Name)_async(@(Model.MethodParameterDeclaration))
   @EmptyLine
 }
 
-  path_template = '@Model.Url'
+  path_template = '@((Model.Url.StartsWith("/") ? Model.Url.TrimStart('/') : (string)Model.Url))'
```
Comment (vishrutshah) [inline src/generator/AutoRest.Ruby/Templates/MethodTemplate.cshtml]: Does this change rely on information that `base_url` where the path is going to be appended must end with `/` ?

Comment (sarangan12) [inline src/generator/AutoRest.Ruby/Templates/MethodTemplate.cshtml]: No. It does not rely on the information

**Comment by sarangan12** [comment]
@vishrutshah Doing it in runtime will not be correct. If we do it in ms_rest, we are restricting the users (who use only ms_rest). i.e. if a user genuinely has both absolute path & relative paths, the user cannot use ms_rest. So, ms_rest should not be playing with the paths. 

Next, think about ms_rest_azure. You can do the change there. But, think of the users who use autorest with AutoRest.Ruby (not AutoRest.RubyAzure). We will come with the same issue. 

So, Yes. We need to do it in every operation file.

**Comment by sarangan12** [comment]
Rel Issue: https://github.com/Azure/azure-sdk-for-net/issues/3007

**Comment by vishrutshah** [comment]
@sarangan12 

Based on our discussion, https://github.com/Azure/autorest/blob/master/src/generator/AutoRest.CSharp/Templates/Rest/Client/MethodTemplate.cshtml#L133 the logic that .net generator is using. Our changes look fine but let's keep an eye on the issue and what's their view.

**Pull request merged**